### PR TITLE
Add org.freedesktop.Sdk.Extension.ldc

### PR DESCRIPTION
**Pull request by Cogitri**
LDC is a compiler for DLang.

Ref #1379

### CODE REVIEW

**Comment by Cogitri** [comment]
The problem about building it with gdc for support on arm is that I can't really build gdc without gcc, so it'd probably be best to just have gdc in the Freedesktop Sdk. I'll build from ldc binaries for now.

**Comment by Cogitri** [comment]
This builds from souce now, thanks @gasinvein

File: org.freedesktop.Sdk.Extension.ldc.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  - name: appdata
+    buildsystem: simple
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/share/metainfo
+      - cp ${FLATPAK_ID}.metainfo.xml ${FLATPAK_DEST}/share/metainfo
+      - appstream-compose --basename=o${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}
+    sources:
+      - type: file
+        path: org.freedesktop.Sdk.Extension.ldc.metainfo.xml
```
Comment (gasinvein) [inline org.freedesktop.Sdk.Extension.ldc.yml]: These both modules are pretty trivial, maybe merge them into one?

File: org.freedesktop.Sdk.Extension.ldc.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          - 'export LIBRARY_PATH="$LIBRARY_PATH:/usr/lib/sdk/ldc/lib"'
+          - 'export LD_LIBRARY_PATH="LD_LIBRARY_PATH:/usr/lib/sdk/ldc/lib"'
+        dest-filename: enable.sh
+
+  - name: appdata
+    buildsystem: simple
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/share/metainfo
+      - cp ${FLATPAK_ID}.metainfo.xml ${FLATPAK_DEST}/share/metainfo
+      - appstream-compose --basename=o${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}
```
Comment (gasinvein) [inline org.freedesktop.Sdk.Extension.ldc.yml]: `o${FLATPAK_ID}` - a typo (note the `o`)?

Comment (Cogitri) [reply org.freedesktop.Sdk.Extension.ldc.yml]: Oh right, good catch

File: org.freedesktop.Sdk.Extension.ldc.yml (hunk 1 above)
Comment (Cogitri) [reply org.freedesktop.Sdk.Extension.ldc.yml]: Ah sure, it just felt like common practice doing it like that looking at other Flatpak manifests.

**Comment by nedrichards** [comment]
bot, build org.freedesktop.Sdk.Extension.ldc

**Comment by Cogitri** [comment]
bot, build org.freedesktop.Sdk.Extension.ldc

**Comment by Cogitri** [comment]
Bringing this to arm would probably be a real pain since that'd need GDC and building GCC in it, so from my side this is good to go.

**Comment by Cogitri** [comment]
One thing that might be a bit of a pain point is that applications will either have to statically link the ldc-runtime or copy the runtime lib/build it themselves (which is probably a good idea anyway, since the runtime breaks compatibility with every release)

**Comment by gasinvein** [comment]
@Cogitri I think it's fine to just copy required at run-time stuff from sdk extension to app bundle. Some other extensions even provide a helper script `install.sh` for this.

**Comment by Cogitri** [comment]
Thanks, added that.

**Comment by Cogitri** [comment]
bot, build org.freedesktop.Sdk.Extension.ldc

**Comment by Cogitri** [comment]
bot, build org.freedesktop.Sdk.Extension.ldc

**Comment by gasinvein** [comment]
@Cogitri Do you plan to submit apps built with this SDK extension?

**Comment by Cogitri** [comment]
Yes, this could be used for Tilix for example.

**Pull request closed without merging**